File: [CHAPTER 8] 다이나믹 프로그래밍/8_1.cpp
// 8_1.cpp
#include <bits/stdc++.h>

using namespace std;

//피보나치 함수를 재귀함수로 표현

int fibo(int x)
{
  if(x==1 || x==2)
  {
    return 1;
  }
  return fibo(x - 1) + fibo(x - 2);
}

int main(void)
{
  cout << fibo(4) << '\n';
}

File: [CHAPTER 8] 다이나믹 프로그래밍/8_1.py
#8_1.py

#피보나치 함수를 재귀함수로 표현

def fibo(x):
  if x == 1 or x == 2:
    #재귀 함수가 종료될수 있는 시점을 명시
    return 1

  return fibo(x - 1) + fibo(x - 2)

print(fibo(4))